toolbox/ci2gubernator/lib/ci2gubernator.go

```diff
@@ -136,14 +136,28 @@ func (c *Converter) UpdateLastBuildTXT() error {
 		return err
 	}
 	if err = m.ContextLock(ctx); err != nil {
+		// force resetting lock state on timeout
+		m.Unlock()
```

Comment:
Is this supposed to try acquire the lock again after force unlocking?
If not, multiple jobs can write at the same time.

However, if you do that, what happens when multiple jobs are waiting and they all try to unlcok and then lock? Are they going to interfere with each other?

Reply:
I am just thinking that the lock is not very necessary in this case. It is ok if the last writer wins, it is eventually consistent. That s how bootstrap.py is doing it.

Reply:
I update the code to retry acquiring the lock after force removal of lock after initial lock() request timeout. I take a look on the implementation of ContextLock() and it is doing retries with exponential backoffs until the given context timeout so once the first failing client remove the lock file, other pending clients who have not return from ContextLock() will retry and get the lock sequentially. 

I would agree with Seb that it is okay for latest-build.txt to be a bit off, since testgrid does not depend on this file it turns out. Only concern is that any future tools that we use might have different assumptions and debugging then would be hard. I guess if the locking scheme becomes an issue again we could go ahead and remove the code.

Reply:
Okay it seems bootstrap.py used by k8s is also assuming last writer wins forever, which means their tooling has to make the same assumption. So I guess it is up to us whether we want to keep the lock.